test/unit/components/uploaderbutton.test.js

```diff
@@ -0,0 +1,48 @@
+/* global chai */
+/* global describe */
+/* global sinon */
+/* global it */
+/* global beforeEach */
+
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+
+import UploaderButton from '../../../app/components/uploaderbutton';
+
+const expect = chai.expect;
+
+describe('UploaderButton', function () {
+  const props = {
+    buttonUrl: 'http://tidepool.org/products/tidepool-uploader/',
+    buttonText: 'Get the Tidepool Uploader',
+    onClick: sinon.spy()
+  };
+
+  let wrapper;
+  beforeEach(() => {
+    wrapper = shallow(
+      <UploaderButton
+        {...props}
+      />
+    );
+  });
+
+  it('should be a function', function() {
+    expect(UploaderButton).to.be.a('function');
+  });
+
+  describe('render', function() {
+    it('should render without problems', function () {
+      expect(wrapper.find(UploaderButton)).to.have.length(0);
+    });
+
+    it('should have provided button text', function () {
+      expect(wrapper.find('.btn-uploader').someWhere(n => (n.text().search(props.buttonText) !== -1))).to.be.true;
+    });
+
+    it('should respond to an onClick event', () => {
```

Comment:
Nitpick, but since sinon spies are stateful I think it's best to include an expectation of a zero callCount before the click simulation. I *think* we tend to do that in this codebase because I think I learned it from Gordon who wrote a lot of the tests in blip...